contracts/airdropsSMCampaign/Existing1155NftDropSMCampaign.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.14;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/AirdropMerkleProof.sol";
+import "../interfaces/AirdropInfoSMCampaign.sol";
+import "../interfaces/IAirBroFactory.sol";
+
+/// @title Airdrops existing ERC1155 tokens for airdrop recipients
+contract Existing1155NftDropSMCampaign is AirdropInfoSMCampaign, AirdropMerkleProof, IERC1155Receiver, Ownable {
+    IERC1155 public immutable rewardedNft;
+    IERC1155 public immutable rewardToken;
+    uint256 public immutable rewardTokenId;
+    uint256 public immutable tokensPerClaim;
+    uint256 public immutable totalAirdropAmount;
+
+    bool public airdropFunded = false;
+    uint256 public airdropFundBlockTimestamp;
+    address internal airdropFundingHolder;
+    address public immutable airBroFactoryAddress;
+
+    event Claimed(address indexed claimer);
+    event AirdropFunded(address contractAddress);
+    event MerkleRootChanged(bytes32 merkleRoot);
+    // event isEligibleForReward(bool);
+
+    error NotOwner();
```

Comment:
Suggestion: You can replace `NotOwner` and `NotAdmin` with `Unauthorized` and use that to save some gas, unless you really need them separate for some reason